Automattic/woocommerce-payments
# BNPL - Fix sending of shipping data for Afterpay in block checkout

Fixes part of #6792 

#### Changes proposed in this Pull Request

Ensure for Afterpay payments, the shipping address is sent when both billing and shipping address are separately mentioned at **Block checkout**. 

#### Testing instructions

1.  Make sure your local setup is connected ot an account with Afterpay enabled
2. Your test site should have a shipping zone setup for US to allow a separate shipping address to appear at checkout. [Quick video](https://github.com/Automattic/woocommerce-payments/assets/4162931/b527908d-bd62-4e75-9893-14381a2bc970) showing the settings
3. Also ensure you have a block based checkout page available for testing
4. Within WooPayments Dev tools make sure to select `Legacy UPE` 
5. Open the site in an incognito window and try to do a checkout in block checkout with Afterpay
6. Add only the visible Shipping address without opting to add a separate billing address
7. Open the browser console to monitor any errors ( console and network tab ) 
8. Complete the checkout and wait for the test redirect screen of Stripe to appear that shows the data submitted in the transaction
9. Within the Payment Intent details on that page, please check the Shipping address in he Payment Intent data, and the billing address in the billing data below. They should be both the shipping address specified.
10. Repeat the test transaction steps - this time - add a separate Billing address.
11. The submitted Payment intent information on the test Afterpay authorization page should show the shipping address, the billing details should show the separate billing address. 
-------------------


**Post merge**
